src/main/java/com/sngular/kloadgen/extractor/extractors/AvroExtractor.java

```diff
@@ -14,15 +14,30 @@
 import org.apache.avro.Schema.Type;
 import org.apache.commons.collections4.IteratorUtils;
 
-public class AvroExtractor {
+public class AvroExtractor implements Extractor {
 
   private final Set<Schema.Type> typesSet = EnumSet.of(Type.INT, Type.DOUBLE, Type.FLOAT, Type.BOOLEAN, Type.STRING, Type.LONG, Type.BYTES, Type.FIXED);
 
   private final RandomObject randomObject = new RandomObject();
 
-  public final List<FieldValueMapping> processSchema(final Schema schema) {
+  @Override
+  public final List<FieldValueMapping> processSchema(final Object schema) {
     final var attributeList = new ArrayList<FieldValueMapping>();
-    schema.getFields().forEach(field -> processField(field, attributeList, true, true));
+    ((AvroSchema) schema).rawSchema().getFields().forEach(field -> processField(field, attributeList, true, true));
+    return attributeList;
+  }
+
+  @Override
+  public final List<FieldValueMapping> processApicurioParsedSchema(final Object schema) {
```

Comment:
The methods processApicurioParsedSchema and processConvluenteParsedSchema have the same implementation except for the type of argument. To avoid casting and because both methods should be technology provider agnostic, I suggest to specialise the input argument called schema and unify both implementations.

Reply:
¿?